annotation-server/src/medications/medications.service.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { ConfigModule } from '@nestjs/config';
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+
+import { Medication } from './medication.entity';
+import { MedicationsService } from './medications.service';
+
+describe('MedicationsService', () => {
+    let medicationsService: MedicationsService;
+
+    beforeEach(async () => {
+        const modelFixture: TestingModule = await Test.createTestingModule({
+            imports: [ConfigModule.forRoot()], // needed for MedicationService
+            providers: [
+                MedicationsService,
+                {
+                    provide: getRepositoryToken(Medication),
+                    useValue: {},
+                },
+            ],
+        }).compile();
+
+        medicationsService =
+            modelFixture.get<MedicationsService>(MedicationsService);
+    });
+
+    describe('getJSONFromZip', () => {
+        it('should return the json path', async () => {
+            const returnString = await medicationsService.getJSONfromZip();
+            expect(returnString).toContain('drugbank-data.json');
```

Comment:
I think it might be safer to just test that any JSON-file's path is returned, e.g. (not tested!):

```suggestion
            expect(returnString).toContain(/\.json$/);
```

(also see comment on test below)